### PR TITLE
Make it so the AI has a notification for when lasers goes out.

### PR DESCRIPTION
**Pull request by lbnesquik**
## About The Pull Request

There is a noise for the OB one.
I'd like to have the CAS flares link directly to the condor, but i am not certain how to do it.

## Changelog
:cl:
expansion: The AI is now notified when a CAS or OB laser is created or goes out.
/:cl:

### CODE REVIEW

**Review by TiviPlus — APPROVED** [review]

**Review by SplinterGP — CHANGES_REQUESTED** [review]
Make it emit a global signal

File: code/__DEFINES/dcs/signals.dm [hunk 1]
```diff
@@ -40,6 +40,10 @@
 ///Sent when the floodlight switch is powered
 #define COMSIG_GLOB_FLOODLIGHT_SWITCH "floodlight_switch_power_change"
 
+//Signals for fire support
+#define COMSIG_GLOB_OB_LASER_SENT "ob_laser_sent"
+#define COMSIG_GLOB_CAS_LASER_SENT "cas_laser_sent"
```
Comment (TiviPlus) [inline code/__DEFINES/dcs/signals.dm:45]:
```suggestion
#define COMSIG_GLOB_FLOODLIGHT_SWITCH "!floodlight_switch_power_change"

//Signals for fire support
#define COMSIG_GLOB_OB_LASER_SENT "!ob_laser_sent"
#define COMSIG_GLOB_CAS_LASER_SENT "!cas_laser_sent"
```

File: code/game/objects/effects/overlays.dm [hunk 2]
```diff
@@ -140,6 +140,9 @@
 	. = ..()
 	linked_cam = new(loc, name)
 	GLOB.active_cas_targets += src
+	SEND_GLOBAL_SIGNAL(COMSIG_GLOB_CAS_LASER_SENT, src)
+
+
```
Comment (TiviPlus) [inline code/game/objects/effects/overlays.dm:145]:
```suggestion
	SEND_GLOBAL_SIGNAL(COMSIG_GLOB_CAS_LASER_SENT, src)
```

File: code/modules/mob/living/silicon/ai/ai.dm [hunk 3]
```diff
@@ -108,6 +112,18 @@
...
 	current_order = order
 
+
+///Receive fire support laser notifications
+/mob/living/silicon/ai/proc/receive_laser_OB(datum/source, obj/effect/overlay/temp/laser_target/OB/incoming_laser)
+	SIGNAL_HANDLER
+	to_chat(src, "<span class='notice'>Orbital Bombardment laser detected. Target: [AREACOORD_NO_Z(incoming_laser)] </span>")
+	playsound(src, 'sound/effects/binoctarget.ogg', 15)
+
+/mob/living/silicon/ai/proc/receive_laser_CAS(datum/source, obj/effect/overlay/temp/laser_target/cas/incoming_laser)
```
Comment (TiviPlus) [inline code/modules/mob/living/silicon/ai/ai.dm:122]: Do not use upper case in proc names like this

File: code/modules/mob/living/silicon/ai/ai.dm [hunk 4]
```diff
@@ -76,6 +76,10 @@
 
 	RegisterSignal(src, COMSIG_MOB_CLICK_ALT, .proc/send_order)
 	RegisterSignal(src, COMSIG_ORDER_SELECTED, .proc/set_order)
+	RegisterSignal(SSdcs, COMSIG_GLOB_OB_LASER_SENT, .proc/receive_laser_OB)
+	RegisterSignal(SSdcs, COMSIG_GLOB_CAS_LASER_SENT, .proc/receive_laser_CAS)
```
Comment (TiviPlus) [inline code/modules/mob/living/silicon/ai/ai.dm:80]: Rename these signals to end with _CREATED instead of _SENT as that is more indicative

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/silicon/ai/ai.dm (hunk 4 above)
Comment (lbnesquik) [reply code/modules/mob/living/silicon/ai/ai.dm]: Applied

File: code/modules/mob/living/silicon/ai/ai.dm (hunk 3 above)
Comment (lbnesquik) [reply code/modules/mob/living/silicon/ai/ai.dm]: Applied

File: code/game/objects/effects/overlays.dm (hunk 2 above)
Comment (lbnesquik) [reply code/game/objects/effects/overlays.dm]: Applied ? If you meant the bad spacing.

File: code/__DEFINES/dcs/signals.dm (hunk 1 above)
Comment (lbnesquik) [reply code/__DEFINES/dcs/signals.dm]: Applied

File: code/modules/mob/living/silicon/ai/ai.dm (hunk 3 above)
Comment (lbnesquik) [reply code/modules/mob/living/silicon/ai/ai.dm]: Applied

**Review by TiviPlus — APPROVED** [review]

File: code/modules/mob/living/silicon/ai/ai.dm [hunk 5]
```diff
@@ -108,6 +114,18 @@
...
+
+///Receive fire support laser notifications
+/mob/living/silicon/ai/proc/receive_laser_ob(datum/source, obj/effect/overlay/temp/laser_target/OB/incoming_laser)
+	SIGNAL_HANDLER
+	to_chat(src, "<span class='notice'>Orbital Bombardment laser detected. Target: [AREACOORD_NO_Z(incoming_laser)] </span>")
+	playsound(src, 'sound/effects/binoctarget.ogg', 15)
+
+/mob/living/silicon/ai/proc/receive_laser_cas(datum/source, obj/effect/overlay/temp/laser_target/cas/incoming_laser)
+	SIGNAL_HANDLER
+	to_chat(src, "<span class='notice'>CAS laser detected. Target: [AREACOORD_NO_Z(src)]</span>")
```
Comment (TiviPlus) [inline code/modules/mob/living/silicon/ai/ai.dm:126]: Ah wait these two playsounds should be playsound local or send_sound

Comment (lbnesquik) [reply code/modules/mob/living/silicon/ai/ai.dm]: Applied

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**